Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h

```diff
@@ -269,7 +269,7 @@ class RigidCoord<3,real>
     void toMatrix( Mat& m) const
     {
         m.identity();
-        orientation.toMatrix(m);
+        orientation.toHomogeneousMatrix(m);
```

Comment:
```suggestion
       if constexpr (std::is_same_v<Mat, sofa::type::Mat<3,3,real>)
       {
            orientation.toMatrix(m);
       }
       else if constexpr (std::is_same_v<Mat, sofa::type::Mat<4,4,real>)
       {
           orientation.toHomogeneousMatrix(m);
       }
       else
       {
           static_assert(false, "toMatrix() only supports sofa::type::Mat<3,3,Real> or sofa::type::Mat<4,4,Real>");
       }
```

YOLO, I did not try to compile it so typos may exist 🤫

Reply:
I did the modifications with some fix.